solr/core/src/java/org/apache/solr/handler/admin/MetricsHandler.java

```diff
@@ -186,10 +187,22 @@ private NamedList<Object> handlePrometheusExport(SolrParams params) {
     List<MetricType> metricTypes = parseMetricTypes(params);
     List<MetricFilter> metricFilters =
         metricTypes.stream().map(MetricType::asMetricFilter).collect(Collectors.toList());
+
     Set<String> requestedRegistries = parseRegistries(params);
+    MetricRegistry mergedCoreRegistries = new MetricRegistry();
 
     for (String registryName : requestedRegistries) {
       MetricRegistry dropwizardRegistry = metricManager.registry(registryName);
+
+      // Merge all core registries into a single registry and
+      // append the core name to the metric to avoid duplicate metrics name
+      if (registryName.startsWith("solr.core")) {
+        mergedCoreRegistries.registerAll(
+            Arrays.stream(registryName.split("\\.")).skip(1).collect(Collectors.joining("_")),
```

Comment:
It'd be faster to do a substring on the index after the first period, and then with the result of that, replace all periods with underscores.  Having a little utility method to do this with comment would ultimately be clearer IMO